### PR TITLE
Role updates

### PR DESCRIPTION
**Pull request by sean-m-sullivan**
# What does this PR do?

Adds two roles and updates options on various roles in order to bring the collection to parity with the awx.awx collection.

# How should this be tested?

Automated tests have been created for the roles. 

Currently working on getting the tests working, but its in a place where its almost done and ready for review.

### CODE REVIEW

**Review by djdanielsson — COMMENTED** [review]
what is the point of bulk hosts? wouldn't it be better to push users into inventory source by file and manage the hosts that way?

**Review by djdanielsson — CHANGES_REQUESTED** [review]
I also am not sure what I think of bulk launching jobs, but that one I am less against then hosts.

File: roles/bulk_host_create/README.md [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+### Authentication
+
+|Variable Name|Default Value|Required|Description|Example|
+|:---|:---:|:---:|:---|:---|
+|`controller_hostname`|""|yes|URL to the Ansible Controller Server.|127.0.0.1|
+|`controller_validate_certs`|`True`|no|Whether or not to validate the Ansible Controller Server's SSL certificate.||
+|`controller_username`|""|no|Admin User on the Ansible Controller Server. Either username / password or oauthtoken need to be specified.||
+|`controller_password`|""|no|Controller Admin User's password on the Ansible Controller Server. This should be stored in an Ansible Vault at vars/controller-secrets.yml or elsewhere and called from a parent playbook. Either username / password or oauthtoken need to be specified.||
+|`controller_oauthtoken`|""|no|Controller Admin User's token on the Ansible Controller Server. This should be stored in an Ansible Vault at or elsewhere and called from a parent playbook. Either username / password or oauthtoken need to be specified.|||
+|`controller_************`|`see below`|yes|Data structure describing your organization or organizations Described below.||
```
Comment (Tompage1994) [inline roles/bulk_host_create/README.md]: The ******* should be filled in as `controller_bulk_hosts`

Comment (Tompage1994) [reply roles/bulk_host_create/README.md]: And the same for all the other instances of ***** below

File: roles/bulk_host_create/meta/argument_specs.yml [hunk 2]
```diff
@@ -0,0 +1,115 @@
+---
+argument_specs:
+  main:
+    short_description: An Ansible Role to run a list of ad hoc commands on Ansible Controller.
+    options:
+      controller_bulk_hosts:
+        description: Data structure describing a bulk of hosts to run
+        type: list
+        elements: dict
+        # options:
```
Comment (Tompage1994) [inline roles/bulk_host_create/meta/argument_specs.yml]: Can you remove this commented section as thats not relevant to this arg_spec.

You could leave in inventory and add hosts if you want but we aren't using it at the moment so it's not really a bother

File: roles/bulk_job_launch/README.md [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+|:---|:---:|:---:|:---|:---|
+|`controller_state`|"present"|no|The state all objects will take unless overridden by object default|'absent'|
+|`controller_hostname`|""|yes|URL to the Ansible Controller Server.|127.0.0.1|
+|`controller_validate_certs`|`True`|no|Whether or not to validate the Ansible Controller Server's SSL certificate.||
+|`controller_username`|""|no|Admin User on the Ansible Controller Server. Either username / password or oauthtoken need to be specified.||
+|`controller_password`|""|no|Controller Admin User's password on the Ansible Controller Server. This should be stored in an Ansible Vault at vars/controller-secrets.yml or elsewhere and called from a parent playbook. Either username / password or oauthtoken need to be specified.||
+|`controller_oauthtoken`|""|no|Controller Admin User's token on the Ansible Controller Server. This should be stored in an Ansible Vault at or elsewhere and called from a parent playbook. Either username / password or oauthtoken need to be specified.|||
+|`controller_************`|`see below`|yes|Data structure describing your organization or organizations Described below.||
```
Comment (Tompage1994) [inline roles/bulk_job_launch/README.md]: Same thing with ****** again

File: roles/bulk_job_launch/meta/argument_specs.yml [hunk 4]
```diff
@@ -0,0 +1,120 @@
+---
+argument_specs:
+  main:
+    short_description: An Ansible Role to run a list of ad hoc commands on Ansible Controller.
+    options:
+      controller_bulk_job_launch_commands:
+        description: Data structure describing your ad hoc commands to run
+        type: list
+        elements: dict
+        # options:
```
Comment (Tompage1994) [inline roles/bulk_job_launch/meta/argument_specs.yml]: Again, please remove commented section unless its specific for this role

File: changelogs/fragments/role_updates.yml [hunk 5]
```diff
@@ -0,0 +1,7 @@
+---
+minor_changes:
+  - Added new_name option to the roles applications, credential_types, execution_environments, inventories, projects, users.
```
Comment (Tompage1994) [inline changelogs/fragments/role_updates.yml:3]: There was also new_username and the changes in workflow_job_templates

**Review by Tompage1994 — CHANGES_REQUESTED** [review]
A few changes needed

**Comment by Tompage1994** [comment]
> what is the point of bulk hosts? wouldn't it be better to push users into inventory source by file and manage the hosts that way?

These are just the new modules in awx.awx > https://github.com/ansible/awx/tree/devel/awx_collection/plugins/modules

**Comment by djdanielsson** [comment]
I will approve once the changes Tom requested are done

File: roles/bulk_host_create/README.md (hunk 1 above)
Comment (sean-m-sullivan) [reply roles/bulk_host_create/README.md]: Fixed

File: roles/bulk_host_create/meta/argument_specs.yml (hunk 2 above)
Comment (sean-m-sullivan) [reply roles/bulk_host_create/meta/argument_specs.yml]: Fixed

File: roles/bulk_job_launch/README.md (hunk 3 above)
Comment (sean-m-sullivan) [reply roles/bulk_job_launch/README.md]: Fixed

File: roles/bulk_job_launch/meta/argument_specs.yml (hunk 4 above)
Comment (sean-m-sullivan) [reply roles/bulk_job_launch/meta/argument_specs.yml]: Fixed

File: changelogs/fragments/role_updates.yml (hunk 5 above)
Comment (sean-m-sullivan) [reply changelogs/fragments/role_updates.yml:3]: Updated

**Review by djdanielsson — APPROVED** [review]

**Pull request merged**